### PR TITLE
[BACKLOG-2540] - AMD Conversion

### PR DESCRIPTION
**Pull request by dcleao**
@pminutillo, @DFieldFL please review.

### CODE REVIEW

**Comment by pminutillo** [comment]
@krivera-pentaho if you get any time, could you take a look at this pull request? Another set of eyes would be appreciated. Thanks.

**Comment by dcleao** [comment]
Just replaced the commit with one with the indentation fixed :-/

**Comment by pminutillo** [comment]
This looks ok to me

File: package-res/resources/web/_activeVizTypes.js [hunk 1]
```diff
@@ -14,9 +14,25 @@
...
  */
-pen.define([
-    'common-ui/vizapi/VizController'
-], function() {
-    // Unfortunately couldn't configure this dependency through a shim and `init`. Didn't run.
-    return pentaho.VizController;
-});
\ No newline at end of file
+define([
+    "require",
```
Comment (krivera-pentaho) [inline package-res/resources/web/_activeVizTypes.js]: If we can "define" already, then why do you have to require "require"?

File: package-res/resources/web/pentaho/common/propertiesPanel/ColorBox.js [hunk 2]
```diff
@@ -14,66 +14,69 @@
...
 
-            if( !ColorDialog.dialog ) {
-                ColorDialog.dialog = new ColorDialog();
-            }
-            var func = lang.hitch( this, function() { this.colorsChanged() } );
-            ColorDialog.dialog.registerOnSuccessCallback( func );
-            ColorDialog.dialog.show();
-        },
+            onClick: function() {
+                if(!ColorDialog.dialog) ColorDialog.dialog = new ColorDialog();
```
Comment (krivera-pentaho) [inline package-res/resources/web/pentaho/common/propertiesPanel/ColorBox.js]: Is there a reason the { } were dropped from the if statement? It is usually bad practice, even for one line if statements, to remove them.

File: package-res/resources/web/pentaho/common/propertiesPanel/ColorBox.js [hunk 3]
```diff
@@ -14,66 +14,69 @@
...
+            registerOkFunc: function(func) {
+                this.okFunc = func;
+            },
+
+            colorsChanged: function() {
+                var color = ColorDialog.dialog.color;
+                if(color != null && color != this.value) {
+                    this.value = color;
+                    style.set(this.colorBox, "background-color", this.value);
+                    if(this.okFunc) this.okFunc(color);
```
Comment (krivera-pentaho) [inline package-res/resources/web/pentaho/common/propertiesPanel/ColorBox.js]: The same comment here for this if statement as well

File: package-res/resources/web/_activeVizTypes.js (hunk 1 above)
Comment (dcleao) [inline package-res/resources/web/_activeVizTypes.js]: That's an interesting one. Requiring `require` returns what is called a contextualized require function. Unlike the global `require` function, that resolves relative paths in relation to `config.baseUrl`, the contextualized one resolves them in relation to where the `define` from which it was required.

File: package-res/resources/web/pentaho/common/propertiesPanel/ColorBox.js (hunk 2 above)
Comment (dcleao) [inline package-res/resources/web/pentaho/common/propertiesPanel/ColorBox.js]: It is a matter of style, of readability. Considered by some a bad practice, but, luckily, not by everyone.

**Comment by krivera-pentaho** [comment]
Everything seems ok for me

**Comment by DFieldFL** [comment]
Thumbs up!

**Pull request merged**